File: mynlp-transform/src/main/java/com/mayabot/nlp/transform/Traditional2Simplified.java
package com.mayabot.nlp.transform;

import javax.inject.Singleton;
import java.util.TreeMap;

/**
 * 繁体转简体的词典
 *
 * @author jimichan
 */
@Singleton
public class Traditional2Simplified extends BaseTransformDictionary {

    public static final String rsName = "ts-dict/t2s.txt";

    public Traditional2Simplified() {
    }

    @Override
    public TreeMap<String, String> loadDictionary() {
        return loadFromResouce(rsName);
    }
}

File: modules/mynlp-transform/src/main/java/com/mayabot/nlp/transform/Traditional2Simplified.kt
package com.mayabot.nlp.transform

import com.mayabot.nlp.injector.Singleton
import java.util.TreeMap

/**
 * 繁体转简体的词典
 *
 * @author jimichan
 */
@Singleton
class Traditional2Simplified : BaseTransformDictionary() {

    override fun loadDictionary(): TreeMap<String, String> {
        return loadFromResource(RS_NAME)
    }

    companion object {

        private val RS_NAME = "ts-dict/t2s.txt"
    }
}
